test/socket_test.exs

```diff
@@ -69,96 +75,112 @@ defmodule KafkaEx.Socket.Test do
-      )
-
-    assert socket.ssl == false
-    KafkaEx.Socket.close(socket)
-  end
+  describe "without SSL socket" do
+    setup do
+      Server.start(3040)
+      {:ok, [port: 3040]}
+    end
 
-  test "send and receive using a non SSL socket", context do
-    {:ok, socket} =
-      KafkaEx.Socket.create(
-        'localhost',
-        context[:port],
-        [:binary, {:packet, 0}, {:active, false}],
-        false
-      )
-
-    KafkaEx.Socket.send(socket, 'ping')
-    assert {:ok, "ping"} == KafkaEx.Socket.recv(socket, 0)
-    KafkaEx.Socket.close(socket)
-  end
+    test "create a non SSL socket", context do
+      {:ok, socket} =
+        KafkaEx.Socket.create(
+          'localhost',
+          context[:port],
+          [:binary, {:packet, 0}],
+          false
+        )
+
+      assert socket.ssl == false
+      KafkaEx.Socket.close(socket)
+    end
 
-  test "retrieve info from a non SSL socket", context do
-    {:ok, socket} =
-      KafkaEx.Socket.create(
-        'localhost',
-        context[:port],
-        [:binary, {:packet, 0}, {:active, false}],
-        false
-      )
-
-    info = KafkaEx.Socket.info(socket)
-    assert info[:name] == 'tcp_inet'
-    KafkaEx.Socket.close(socket)
-    assert {:error, :closed} == KafkaEx.Socket.send(socket, 'ping')
-  end
+    test "send and receive using a non SSL socket", context do
+      {:ok, socket} =
+        KafkaEx.Socket.create(
+          'localhost',
+          context[:port],
+          [:binary, {:packet, 0}, {:active, false}],
+          false
+        )
+
+      KafkaEx.Socket.send(socket, 'ping')
+      assert {:ok, "ping"} == KafkaEx.Socket.recv(socket, 0)
+      KafkaEx.Socket.close(socket)
+    end
 
-  test "create a SSL socket", context do
-    {:ok, socket} =
-      KafkaEx.Socket.create(
-        'localhost',
-        context[:ssl_port],
-        [:binary, {:packet, 0}],
-        true
-      )
-
-    assert socket.ssl == true
-    KafkaEx.Socket.close(socket)
+    test "retrieve info from a non SSL socket", context do
+      {:ok, socket} =
+        KafkaEx.Socket.create(
+          'localhost',
+          context[:port],
+          [:binary, {:packet, 0}, {:active, false}],
+          false
+        )
+
+      info = KafkaEx.Socket.info(socket)
+      assert info[:name] == 'tcp_inet'
+      KafkaEx.Socket.close(socket)
+      assert {:error, :closed} == KafkaEx.Socket.send(socket, 'ping')
+    end
   end
 
-  test "send and receive using a SSL socket", context do
-    {:ok, socket} =
-      KafkaEx.Socket.create(
-        'localhost',
-        context[:ssl_port],
-        [:binary, {:packet, 0}, {:active, false}],
-        true
-      )
-
-    KafkaEx.Socket.send(socket, 'ping')
-    assert {:ok, "ping"} == KafkaEx.Socket.recv(socket, 0)
-    KafkaEx.Socket.close(socket)
-  end
+  describe "with ssl socket" do
+    setup do
+      SSLServer.start(3030)
+      {:ok, [ssl_port: 3030]}
+    end
+
+    test "create a SSL socket", context do
+      {:ok, socket} =
```

Comment:
Huh it looks like what happened here is the test failed but then it succeeded on the second run.  I wonder if there's a way to clear these when something like that happens.  I'm not 100% that's even the right thing to do but I'm curious if that's an option.

Reply:
I'm not sure. The socket was closed too fast, but it never happened to me in the local setup.